docs/source/roadmap.rst

```diff
@@ -0,0 +1,162 @@
+you can refer to by name or get one by default based on the passed geometries.
+This registry of resamplers will also make it easier for users or third-party
+libraries to add their own resamplers.
+
+We hope with this basic creation process that we can have more control over
+what algorithms support what features and let the user know when something
+isn't allowed early on with clear error messages. For example, what
+combinations of geometry types are supported by the resampler or what types
+of arrays (xarray.DataArray, dask, or numpy) can be provided.
+
+Using Resamplers
+****************
+
+Once you have your resampler instance you can resample your data by doing:
+
+.. code-block:: python
+
+    new_data = resampler.resample(data, **kwargs)
+
+That's it. There are of course a lot of options hidden in the ``**kwargs``,
+but those will be specific to each algorithm. Our hope is that any
+optimizations or conversions that need to happen to get your data resampled
+can all be contained in these resampler objects and hopefully require less
+from the user.
+
+Alternatively to the ``.resample`` call, users can first call two methods:
+
+.. code-block:: python
+
+    resampler.precompute()
+    new_data = resampler.compute(data, **kwargs)
+
+This ``precompute`` method will perform any computations that can be done
+without needing the actual "image" data. You can then call ``.compute``
+to do the actual resampling. This separation is important when we start talking
+about caching (see below).
+
+Caching
+*******
+
+One major simplification we're hoping to achieve with Pyresample 2.0 is a
+defined set of caching functionality all encapsulated in "Cache" objects.
+These objects can be passed to ``create_resampler`` to enable the resampler
+to store intermediate computation results for reuse. How and where that storing
+is done is up to the specific cache object. It could be in-memory only, or
+to zarr datasets on local disk, or to some remote storage.
+
+By calling the ``.precompute`` method, the user will be able to pre-fill this
+cache without needing any image data. This will be useful for users using
+pyresample in operations where they may want to manually fill the cache before
+spawning realtime (time sensitive) processing.
+
+Indexes
+^^^^^^^
+
+From our survey we learned that a lot of users use the indexes returned by
+``get_neighbour_info`` for their own custom analyses. We recognize this need
+and while Cache objects could be written to get the same result, we think
+there is a better way. We'd like to explore the idea of "Index" generating
+objects. Like Resamplers, these would provide you a way of relating a source
+geometry to a destination geometry. However, these objects would only be
+responsible for returning the index information.
+
+We haven't defined the interface for these yet, but hope that having something
+separate from resamplers will serve more people.
```

Comment:
I'd like to add here something to make clear that the feature will be there. Right now I understand that we will do exploration but it doesn't sound like we are sure that the feature will be there.

So maybe:
```
From our survey we learned that a lot of users use the indexes returned by
``get_neighbour_info`` for their own custom analyses. We recognize this need
and will implement it. While Cache objects could be written to get the same result, [...]
```

Reply:
I see what you're saying. Let me see if I can reword it a little bit more.

Reply:
Ok, latest commit should address this and make it a little clearer...I hope.

Reply:
works